R/PipeOpDateFeatures.R

```diff
@@ -0,0 +1,209 @@
+#' added to the feature set of the output task.
+#'
+#' @section Construction:
+#' ```
+#' PipeOpDateFeatures$new(id = "datefeatures", param_vals = list())
+#' ```
+#' * `id` :: `character(1)`\cr
+#'   Identifier of resulting object, default `"datefeatures"`.
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings that would otherwise
+#'   be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [`PipeOpTaskPreprocSimple`].
+#'
+#' The output is the input [`Task`][mlr3::Task] with date-related features computed and added to the
+#' feature set of the output task and the specified `POSIXct` column of the data removed from the
+#' feature set.
+#'
+#' @section State:
+#' The `$state` is a named `list` with the `$state` elements inherited from
+#' [`PipeOpTaskPreprocSimple`], as well as:
+#' * `date_variable` the name of the `POSIXct` column that was used for the feature engineering.
+#' * `cyclic` logical value indicating whether cyclic features were computed.
+#' * `features` a vector of the feature names that were computed.
+#'
+#' @section Parameters:
+#' The parameters are the parameters inherited from [`PipeOpTaskPreprocSimple`], as well as:
+#' * `date_variable` :: `character(1)`\cr
+#'   The name of the `POSIXct` column of the data that should be used for the feature engineering.
+#'   Must be specified during construction.
+#' * `cyclic` :: logical(1)`\cr
+#'   Whether cyclic features should be computed. See Internals. Default FALSE.
+#' * `year` :: `logical(1)`\cr
+#'   Whether the year should be extracted as a feature. Default TRUE.
+#' * `month` :: `logical(1)`\cr
+#'   Whether the month should be extracted as a feature. Default TRUE.
+#' * `week_of_year` :: `logical(1)`\cr
+#'   Whether the week of the year should be extracted as a feature. Default TRUE.
+#' * `day_of_year` :: `logical(1)`\cr
+#'   Whether the day of the year should be extracted as a feature. Default TRUE.
+#' * `day_of_month` :: `logical(1)`\cr
+#'   Whether the day of the month should be extracted as a feature. Default TRUE.
+#' * `day_of_week` :: `logical(1)`\cr
+#'   Whether the day of the week should be extracted as a feature. Default TRUE.
+#' * `hour` :: `logical(1)`\cr
+#'   Whether the hour should be extracted as a feature. Default TRUE.
+#' * `minute` :: `logical(1)`\cr
+#'   Whether the minute should be extracted as a feature. Default TRUE.
+#' * `second` :: `logical(1)`\cr
+#'   Whether the second should be extracted as a feature. Default TRUE.
+#' * `is_day` :: `logical(1)`\cr
+#'   Whether a feature should be extracted indicating whether it is day time (06:00 am - 08:00pm).
+#'   Default TRUE.
+#'
+#' @section Internals:
+#' If `cyclic = TRUE`, cyclic features are computed for the features `"month"`, `"week_of_year"`,
```

Comment:
are computed ... ? in addition to ? ...

Reply:
So currently, we overwrite the date features when we compute cyclic features?
- Is this how it is done in fast.ai?
- Do we want to keep this?

Reply:
I changed this in the updated version so that cyclic features are kept as additional ones.